org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapter.java

```diff
@@ -782,4 +786,180 @@ private String convertDebuggerSourceToClient(String uri) {
     private Types.Message convertDebuggerMessageToClient(String message) {
         return new Types.Message(this.messageId.getAndIncrement(), message);
     }
+
+    private class VariableHandler {
+        private boolean showStaticVariables = true;
+        private IVariableFormatter variableFormatter;
+        private RecyclableObjectPool<ThreadReference, Object> objectPool;
+        private Map<String, Object> options;
+        
+        public VariableHandler(IVariableFormatter variableFormatter, boolean showStaticVariables,
+                               boolean hexFormat, boolean showQualified) {
+            this.objectPool = new RecyclableObjectPool<>();
+            this.variableFormatter = variableFormatter;
+            this.showStaticVariables = showStaticVariables;
+
+            this.options = new HashMap<>();
+            if (hexFormat) {
+                options.put(NumericFormatter.NUMERIC_FORMAT_OPTION, NumericFormatEnum.HEX);
```

Comment:
This is not a one-time setting. A user may change the numeric format to different values on the fly.